### PR TITLE
http1: allow duplicated chunked header

### PR DESCRIPTION
**Pull request by lambdai**
Signed-off-by: Yuchen Dai <silentdai@gmail.com>
Commit Message:
Allow `transfer-encoding: chunked` appears repeatedly in the header.

Additional Description:
Initially reported in https://github.com/istio/istio/issues/24753
I don't find RFC declare duplicated chunked as a failure.
I also confirmed by feeding curl with the duplicated `transfer-encoding: chunked`. No error is reported.

Risk Level: LOW
Testing: unit test
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/common/http/http1/codec_impl_test.cc [hunk 1]
```diff
@@ -2251,6 +2251,28 @@ TEST_F(Http1ClientConnectionImplTest, NoContentLengthResponse) {
...
+  initialize();
+
+  NiceMock<MockResponseDecoder> response_decoder;
+  Http::RequestEncoder& request_encoder = codec_->newStream(response_decoder);
+  TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/"}, {":authority", "host"}};
+  request_encoder.encodeHeaders(headers, true);
+
+  Buffer::OwnedImpl response("HTTP/1.1 200 OK\r\n"
+                             // First chunked header.
+                             "transfer-encoding: chunked\r\n"
```
Comment (alyssawilk) [inline test/common/http/http1/codec_impl_test.cc:2264]: @PiotrSikora for thoughts

I don't *think* this is legal.  Transfer encoding headers are supposed to indicate transformations to the body content

https://tools.ietf.org/html/rfc7230#section-3.3.1

if you send chunked multiple times, you're indicating that the body has been chunked multiple times which is strictly disallowed.  

A sender MUST NOT apply chunked more than once to a message body (i.e., chunking an  already chunked message is not allowed).

Comment (PiotrSikora) [reply test/common/http/http1/codec_impl_test.cc:2264]: I agree, I don't think this is allowed.

**Comment by lambdai** [comment]
I agree "chunked, chunked" is an error. So envoy is doing nothing wrong.

So the question is do we allow "chunked" body and sanitize "chunked, chunked" to "chunked"?

Happy to close if we don't need such a sanitize.

**Comment by PiotrSikora** [comment]
> So the question is do we allow "chunked" body and sanitize "chunked, chunked" to "chunked"?

That would be the one option... but we don't know if that's really the case and if backend simply emitted duplicate headers and response is chunked once, or if response is incorrectly chunked twice (e.g. by multiple layers of HTTP proxies).

As such, I believe that the current behavior of rejecting responses with multiple `transfer-encoding` values is much safer.

**Comment by lambdai** [comment]
> but we don't know if that's really the case and if backend simply emitted duplicate headers and response is chunked once, or if response is incorrectly chunked twice (e.g. by multiple layers of HTTP proxies).

Make sense. Thanks @alyssawilk @PiotrSikora !

**Pull request closed without merging**